doc/adr/0004-nats-headers.md

```diff
@@ -18,7 +18,17 @@ The salient points of the HTTP header specification are:
 - The specification may allow any number of strange things like comments/tokens etc.
 - The keys can repeat.
 
-The reference NATS Go client uses the `http.Header` infrastructure for serializing NATS headers. Other language clients may not be able to reuse header serialization infrastructure outside of the context of an HTTP request. Furthermore, the implementation of the native header library may diverge from the Go `http.Header` while still being standard. 
+More specifically from [rfc822](https://www.ietf.org/rfc/rfc822.txt) Section 3.1.2:
+> Once a field has been unfolded, it may be viewed as being composed of a field-name followed by a colon (":"), followed by a field-body, and  terminated  by  a  carriage-return/line-feed.
+> The  field-name must be composed of printable ASCII characters (i.e., characters that  have  values  between  33.  and  126., decimal, except colon).  The field-body may be composed of any ASCII characters, except CR or LF.  (While CR and/or LF may be present  in the actual text, they are removed by the action of unfolding the field.)
+
+The reference NATS Go client uses the `http.Header` infrastructure for serializing NATS headers. Other language clients may not be able to reuse header serialization infrastructure outside of the context of an HTTP request. Furthermore, the implementation of the native header library may diverge from the Go `http.Header` while still being standard.
```

Comment:
Actually Go client uses `MIMEHeader` to decode/encode headers. This also means different handling of key/value syntax that maybe http header RFC that you point to.